config/main.py

```diff
@@ -5422,6 +5422,97 @@ def disable_use_link_local_only(ctx, interface_name):
     interface_dict = db.get_table(interface_type)
     set_ipv6_link_local_only_on_interface(db, interface_dict, interface_type, interface_name, "disable")
 
+#
+# 'damping' subgroup ('config interface damping ...')
+#
+
+@interface.group(cls=clicommon.AbbreviationGroup)
+@click.pass_context
+def damping(ctx):
+    """Set interface damping configurations"""
+    pass
+
+#
+# 'algo' subcommand ('config interface damping algo ...')
+#
+
+@damping.command()
+@click.pass_context
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+@click.argument('algo_type', metavar='<algo_type>', required=True, type=click.Choice(["aied", "disabled"]))
+def algo(ctx, interface_name, algo_type):
+    """Set link event damping algorithm"""
+    # Get the config_db connector
+    config_db = ctx.obj['config_db']
+
+    if clicommon.get_interface_naming_mode() == "alias":
+        interface_name = interface_alias_to_name(config_db, interface_name)
+        if interface_name is None:
+            ctx.fail("'interface_name' is None!")
+
+    port_dict = config_db.get_table('PORT')
+    if interface_name not in port_dict:
+        ctx.fail("Invalid port {}".format(interface_name))
+
+    log.log_info("Executing: interface link_event_damping_algorithm {} {}".format(interface_name, algo_type))
+
+    config_db.mod_entry("PORT", interface_name, {"link_event_damping_algorithm": algo_type})
+
+#
+# 'aied-param' subcommand ('config interface damping aied-param ...')
+#
+
+@damping.command()
+@click.pass_context
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+@click.option('--max-suppress-time', required=False, type=int, help="Set max suppress time in ms")
+@click.option('--decay-half-life', required=False, type=int, help="Set decay half life in ms")
+@click.option('--suppress-threshold', required=False, type=int, help="Set suppress threshold")
+@click.option('--reuse-threshold', required=False, type=int, help="Set reuse threshold")
+@click.option('--flap-penalty', required=False, type=int, help="Set flap penalty")
+def aied_param(ctx, interface_name, max_suppress_time, decay_half_life, suppress_threshold, reuse_threshold, flap_penalty):
+    """Set AIED link event damping configuration"""
+    # Get the config_db connector
+    config_db = ctx.obj['config_db']
+
+    if clicommon.get_interface_naming_mode() == "alias":
+        interface_name = interface_alias_to_name(config_db, interface_name)
+        if interface_name is None:
+            ctx.fail("'interface_name' is None!")
+
+    port_dict = config_db.get_table('PORT')
+    if interface_name not in port_dict:
+        ctx.fail("Invalid port {}".format(interface_name))
+
+    config_set = {}
+
+    if max_suppress_time is not None:
+        if max_suppress_time < 0:
```

Comment:
could you please use click.IntRange for such validation? check https://click.palletsprojects.com/en/8.1.x/api/#click.IntRange